yasa/evaluation.py

```diff
@@ -0,0 +1,649 @@
+            else:
+                hypno_test = Hypnogram(hypno_test.hypno[:n_ref], n_stages=hypno_test.n_stages)
+                n_trimmed = n_test - n_ref
+                warn_msg = f"`hypno_test` longer than `hypno_ref`, {n_trimmed} epochs trimmed"
+            ## Q: Should be downplayed as INFO?
+            logger.warning(warn_msg)
+        self.hypno_ref = hypno_ref
+        self.hypno_test = hypno_test
+
+    def get_agreement(self):
+        """
+        Return a dataframe of ``hypno_ref``/``hypno_test`` performance
+        across all stages as measured by common classifier agreement methods.
+
+        ## Q: Are there better names to differentiate get_agreement vs get_agreement_by_stage?
+        ##    Maybe should be binary vs multiclass?
+        .. seealso:: :py:meth:`yasa.EpochByEpochResults.get_agreement_by_stage`
+
+        Returns
+        -------
+        agreement : :py:class:`pandas.Series`
+            A :py:class:`pandas.Series` with agreement metrics as indices.
+        """
+        true = self.hypno_ref.hypno.to_numpy()
+        pred = self.hypno_test.hypno.to_numpy()
+        accuracy = metrics.accuracy_score(true, pred)
+        kappa = metrics.cohen_kappa_score(true, pred)
+        jaccard = metrics.jaccard_score(true, pred, average="weighted")
+        precision = metrics.precision_score(true, pred, average="weighted", zero_division=0)
+        recall = metrics.recall_score(true, pred, average="weighted", zero_division=0)
+        f1 = metrics.f1_score(true, pred, average="weighted", zero_division=0)
+        scores = {
+            "accuracy": accuracy,
+            "kappa": kappa,
+            "weighted_jaccard": jaccard,
+            "weighted_precision": precision,
+            "weighted_recall": recall,
+            "weighted_f1": f1,
+        }
+        agreement = pd.Series(scores, name="agreement").rename_axis("metric")
+        return agreement
+
+    def get_agreement_by_stage(self):
+        """
+        Return a dataframe of ``hypno_ref``/``hypno_test`` performance
+        for each stage as measured by common classifier agreement methods.
+
+        .. seealso:: :py:meth:`yasa.EpochByEpochResults.get_agreement`
+
+        Returns
+        -------
+        agreement : :py:class:`pandas.DataFrame`
+            A DataFrame with agreement metrics as indices and stages as columns.
+        """
+        true = self.hypno_ref.hypno.to_numpy()
+        pred = self.hypno_test.hypno.to_numpy()
+        labels = self.hypno_ref.labels  # equivalent to hypno_test.labels
+        scores = metrics.precision_recall_fscore_support(
+            true, pred, labels=labels, average=None, zero_division=0
+        )
+        agreement = pd.DataFrame(scores)
+        agreement.index = pd.Index(["precision", "recall", "fscore", "support"], name="metric")
+        agreement.columns = pd.Index(labels, name="stage")
+        return agreement
+
+    def get_confusion_matrix(self):
```

Comment:
I would add a `normalize=[None, "test", "ref"]` parameter, similar to https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html.

And then add in the documentation that normalization by "ref" means that you are essentially looking at the sensitivity of "test" against "ref", while normalization by "test" means that you are looking at the precision  of "test" against "ref" (right?)

Reply:
Great! Thanks for pointing this out, we're just using `sklearn.metrics.confusion_matrix` directly now (instead of `pandas.crosstab`). Users can pass kwargs to `confusion_matrix`, including `normalize`, and can refer to that documentation.

Reply:
Perfect